weio/fast_output_file.py

```diff
@@ -113,52 +125,92 @@ def readline(iLine):
         else:
-            cols=info['attribute_names']
+            # Normal mode: read entire file
+            try:
+                if ext in ['.out','.elev','.dbg','.dbg2']:
+                    self.data, info = load_ascii_output(self.filename, **kwargs)
+                elif ext=='.outb':
+                    self.data, info = load_binary_output(self.filename, **kwargs)
+                    self['binary']=True
+                elif ext=='.elm':
+                    F=CSVFile(filename=self.filename, sep=' ', commentLines=[0,2],colNamesLine=1)
+                    self.data = F.data
+                    del F
+                    info['attribute_units']=readline(3).replace('sec','s').split()
+                    info['attribute_names']=self.data.columns.values
+                else:
+                    if isBinary(self.filename):
+                        self.data, info = load_binary_output(self.filename, **kwargs)
+                        self['binary']=True
+                    else:
+                        self.data, info = load_ascii_output(self.filename, **kwargs)
+                        self['binary']=False
+            except MemoryError as e:
+                raise BrokenReaderError('FAST Out File {}: Memory error encountered\n{}'.format(self.filename,e))
+            except Exception as e:
+                raise WrongFormatError('FAST Out File {}: {}'.format(self.filename,e.args))
+            if self.data.shape[0]==0:
+                raise EmptyFileError('This FAST output file contains no data: {}'.format(self.filename))
+
+
+
+        # --- Store header information
+        self['attribute_names'] = info.get('attribute_names', [])
+        self['attribute_units'] = info.get('attribute_units', [])
         self.description = info.get('description', '')
         self.description = ''.join(self.description) if isinstance(self.description,list) else self.description
-        if isinstance(self.data, pd.DataFrame):
-            self.data.columns = cols
-        else:
-            if len(cols)!=self.data.shape[1]:
-                raise BrokenFormatError('Inconstistent number of columns between headers ({}) and data ({}) for file {}'.format(len(cols), self.data.shape[1], self.filename))
-            self.data = pd.DataFrame(data=self.data, columns=cols)
+
+        # Store binary file metadata for streaming mode
+        if self.streaming and self['binary']:
+            self['_FileID'] = info.get('FileID')
+            self['_NumOutChans'] = info.get('NumOutChans')
+            self['_NT'] = info.get('NT')
+            self['_ColScl'] = info.get('ColScl')
+            self['_ColOff'] = info.get('ColOff')
+            if info['FileID'] == FileFmtID_WithTime:
+                self['_TimeScl'] = info.get('TimeScl')
+                self['_TimeOff'] = info.get('TimeOff')
+            else:
+                self['_TimeOut1'] = info.get('TimeOut1')
+                self['_TimeIncr'] = info.get('TimeIncr')
+

```

Comment:
the lines above seem complicated, we can just store info into the object self['header_info'] = info